metis-common/src/main/java/eu/europeana/metis/utils/MongoRequestUtil.java

```diff
@@ -0,0 +1,89 @@
+package eu.europeana.metis.utils;
+
+import com.mongodb.MongoSocketException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Simon Tzanakis (Simon.Tzanakis@europeana.eu)
+ * @since 2018-07-24
+ */
+public final class MongoRequestUtil {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongoRequestUtil.class);
+  private static final int MAX_RETRIES = 10;
+  private static final int SLEEP_TIMEOUT = 500;
+
+  private MongoRequestUtil() {
+  }
+
+  /**
+   * Retries a request to Mongo in case an exception with root cause message "Connection reset"
+   * occurred.
+   *
+   * @param supplierWithMongoSocketException the respective supplier encapsulating the Mongo
+   * request
+   * @return the expected object as a result of the mongo request
+   */
+  public static <R> R retryableMongoRequest(
+      SupplierWithMongoSocketException<R> supplierWithMongoSocketException) {
+    int retryCounter = 0;
+
+    R result = null;
+    boolean success = false;
+    do {
+      try {
+        result = supplierWithMongoSocketException.get();
```

Comment:
Just do:
return supplierWithMongoSocketException.get();
No need to keep a boolean as well as the result in a variable.
The loop should be do ... while (true).